# TESTING OF HYPOTHESIS

The question is:

    
    
   A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it.
     
     
     
     
     
     
   ● The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
  
  ● The company needs to check whether there is any dependency between the features “Region” and “Manager”.
   
   Help the company to carry out their study with the help of data provided.

The given dataset consists of 22 rows and 5 columns namely Month, Region, Manager, Sales_before_digital_add(in $) and 

Sales_After_digital_add(in $).

In [ ]:
import numpy as np
import pandas as pd
import seaborn as sns
import scipy.stats as stats

In [ ]:
data=pd.read_csv("Sales_add.csv")

In [ ]:
data.head()

In [ ]:
data.shape

In [ ]:
data.info()

In [ ]:
data.isna().sum()

Hence, no null values present in the dataset.

# 1. Whether there is any increase in sales after stepping into digital marketing

In [25]:
data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

We have to check whether there is any increase in sales after stepping into digital marketing . For that we should carry out one-tailed t-test such that

NULL HYPOTHESIS: Average sales before and after digital marketing are same against

ALTERNATIVE HYPOTHESIS:Average sales after digital marketing is greater than before digital marketing

Next, consider the p-value,

If the p-value is less than your significance level, you can reject the null hypothesis and conclude that the effect is statistically significant. In other words, the evidence in your sample is strong enough to be able to reject the null hypothesis at the population level.

Here, the significane level is assumed to be 0.05.

In [27]:
ttest,pval = stats.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])
print(pval)

6.336667004575778e-11


INSIGHT: Here, the p-value, 6.336667004575778e-11, is less than the significance level, 0.05.

Hence, we should reject the null hypothesis and so accept alternative hypothesis.

So, we can conclude that the average sales after digital marketing is greater than before digital marketing.

Hence, **there is an increase in sales after stepping into digital marketing.**

# 2. Whether there is any dependency between the features “Region” and “Manager”

We have to check whether there is any dependency between the features “Region” and “Manager”. For that we should carry out chi-square test for independence such that

NULL HYPOTHESIS:There is dependency between the features “Region” and “Manager” against

ALTERNATIVE HYPOTHESIS:There is not any dependency between the features “Region” and “Manager”



In [34]:
data_chi = pd.read_csv('Sales_add.csv')
contingency_table=pd.crosstab(data_chi["Region"],data_chi["Manager"])
print('contingency_table :-\n',contingency_table)

contingency_table :-
 Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1


In [35]:
#Observed Values
Observed_Values = contingency_table.values 
print("Observed Values :-\n",Observed_Values)

Observed Values :-
 [[4 3 3]
 [4 1 2]
 [1 3 1]]


In [36]:
b=stats.chi2_contingency(contingency_table)
Expected_Values = b[3]
print("Expected Values :-\n",Expected_Values)

Expected Values :-
 [[4.09090909 3.18181818 2.72727273]
 [2.86363636 2.22727273 1.90909091]
 [2.04545455 1.59090909 1.36363636]]


In [43]:
no_of_rows=len(contingency_table.iloc[0:3,0])
no_of_columns=len(contingency_table.iloc[0,0:3])
ddof=(no_of_rows-1)*(no_of_columns-1)
print("Degree of Freedom:-",ddof)
alpha = 0.05

Degree of Freedom:- 4


In [48]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(Observed_Values,Expected_Values)])
chi_square_statistic=chi_square[0]+chi_square[1]
print("chi-square statistic:-",chi_square_statistic)

chi-square statistic:- 2.921995464852608


In [49]:
critical_value=chi2.ppf(q=1-alpha,df=ddof)
print('critical_value:',critical_value)

critical_value: 9.487729036781154


In [50]:
#p-value
p_value=1-chi2.cdf(x=chi_square_statistic,df=ddof)
print('p-value:',p_value)

p-value: 0.5709629929220089


In [51]:
print('Significance level: ',alpha)
print('Degree of Freedom: ',ddof)
print('chi-square statistic:',chi_square_statistic)
print('critical_value:',critical_value)
print('p-value:',p_value)

Significance level:  0.05
Degree of Freedom:  4
chi-square statistic: 2.921995464852608
critical_value: 9.487729036781154
p-value: 0.5709629929220089


if chi-square_statistic>=critical_value:
    Reject null hypothesis, ie; There is not any relationship between 2 categorical variables
else:
    Accept null hypothesis, ie; There is a relationship between 2 categorical variables
    
    
if p_value<=alpha:
    Reject null hypothesis, ie; There is not any relationship between 2 categorical variables
else:
    Accept null hypothesis, ie; There is a relationship between 2 categorical variables    

INSIGHT:-
Here, the chi-square statistic, 2.921995464852608, is less than the critical value, 9.487729036781154 , so, we accept the null hypothesis

OR

We can consider the p-value, 0.5709629929220089, which is greater than the significance level, 0.05, we accept the null hypothesis.

Hence, we can conclude that **there is a dependency between the  features “Region” and “Manager”**.

ie; there is an association between “Region” and “Manager”.